tests/unit/service_test.py

```diff
@@ -150,6 +150,40 @@ def test_build_port_bindings_with_nonmatching_internal_ports(self):
         self.assertEqual(port_bindings["1000"], [("127.0.0.1", "1000")])
         self.assertEqual(port_bindings["2000"], [("127.0.0.1", "2000")])
 
+    def test_specifies_host_port_with_host_ip_and_port(self):
+        service = Service(
+            'foo',
+            image='foo',
+            ports=["127.0.0.1:1000:2000"])
+        self.assertEqual(service.specifies_host_port(), True)
```

Comment:
Minor nit, now that we're using `py.test` we actually get better error messages if you use `assert service.specifies_host_port()` or `assert not ...`.

Would you mind changing the assertions to use bare `assert` ?